### PR TITLE
Reorganize / split EDMF from dycore

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR:
 - Moves all of the callbacks from `driver/main.jl` into a separate file: `driver/callbacks.jl`
 - Moves all of the specified field variable names, specific to edmf, into `src/variables.jl`
 - Moves the `src/Variables.jl` `satadjust` method into `driver/dycore.jl`

### CODE REVIEW

File: driver/dycore.jl [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# Face only
+face_prognostic_vars(FT, n_up) = (; w = FT(0), TC.face_prognostic_vars_edmf(FT, n_up)...)
+# TC.face_prognostic_vars_edmf(FT, n_up) = (;) # could also use this for empty model
+
+
+#####
+##### Methods
+#####
+
+function satadjust(gm::TC.GridMeanVariables, grid, state)
```
Comment (yairchn) [inline driver/dycore.jl:77]: if this function is used only in initialization I believe we can / should just added it to"initial_conditions.jl" or "Cases.jl"

**Review by yairchn — COMMENTED** [review]
the PR is not working yet - not sure why - but the general direction is great, left one comment

File: driver/dycore.jl (hunk 1 above)
Comment (charleskawczynski) [reply driver/dycore.jl:77]: Ah, good point, I'm fine with that, thanks!

**Comment by yairchn** [comment]
Bors r+

**Pull request merged**